Explorer/Assets/DCL/Web3/Authenticators/DappWeb3Authenticator.cs

```diff
@@ -14,83 +13,122 @@
 
         public DappWeb3Authenticator(IWebBrowser webBrowser,
-            string serverUrl,
-            string signatureUrl,
+            URLAddress authApiUrl,
+            URLAddress signatureWebAppUrl,
+            URLDomain rpcServerUrl,
             IWeb3IdentityCache identityCache,
             IWeb3AccountFactory web3AccountFactory,
-            HashSet<string> whitelistMethods
-        )
+            HashSet<string> whitelistMethods,
+            HashSet<string> readOnlyMethods,
+            DecentralandEnvironment environment)
         {
             this.webBrowser = webBrowser;
-            this.serverUrl = serverUrl;
-            this.signatureUrl = signatureUrl;
+            this.authApiUrl = authApiUrl;
+            this.signatureWebAppUrl = signatureWebAppUrl;
+            this.rpcServerUrl = rpcServerUrl;
             this.identityCache = identityCache;
             this.web3AccountFactory = web3AccountFactory;
             this.whitelistMethods = whitelistMethods;
+            this.readOnlyMethods = readOnlyMethods;
+            this.environment = environment;
         }
 
         public void Dispose()
         {
-            try { webSocket?.Dispose(); }
+            try { authApiWebSocket?.Dispose(); }
             catch (ObjectDisposedException) { }
         }
 
-        public async UniTask<T> SendAsync<T>(EthApiRequest request, CancellationToken ct)
+        public async UniTask<EthApiResponse> SendAsync(EthApiRequest request, CancellationToken ct)
         {
             if (!whitelistMethods.Contains(request.method))
                 throw new Web3Exception($"The method is not allowed: {request.method}");
 
-            try
+            if (string.Equals(request.method, "eth_accounts")
+                || string.Equals(request.method, "eth_requestAccounts"))
             {
-                await ConnectToServerAsync();
+                string[] accounts = Array.Empty<string>();
+
+                if (identityCache.Identity != null)
+                    accounts = new string[] { identityCache.EnsuredIdentity().Address };
 
-                SignatureIdResponse authenticationResponse = await RequestEthMethodAsync(new AuthorizedEthApiRequest
+                return new EthApiResponse
                 {
-                    method = request.method,
-                    @params = request.@params,
-                    authChain = identityCache.Identity!.AuthChain.ToArray(),
-                }, ct);
+                    id = request.id,
+                    jsonrpc = "2.0",
+                    result = accounts,
+                };
+            }
 
-                DateTime signatureExpiration = DateTime.UtcNow.AddMinutes(5);
+            if (string.Equals(request.method, "eth_chainId"))
+            {
+                // TODO: this is a temporary thing until we solve the network in a better way
+                string chainId = environment == DecentralandEnvironment.Org ? MAINNET_CHAIN_ID : SEPOLIA_CHAIN_ID;
```

Comment:
We could isolate the chainID and network calculations into 2 methods (like `GetChainID() and `GetNetworkId()` or similar) so that it's easier to just update those 2 in the future if we want to change that logic.

WDYT?

Reply:
Agree!